src/main/java/org/takes/rq/RqLive.java

```diff
@@ -132,6 +104,28 @@ public InputStream body() {
     }
 
     /**
+     * Checks whether or not the next byte to read is a Line Feed.
+     * @param input The input stream to read
+     * @param baos Current read header
+     * @param position Header line number
+     * @throws IOException If the next byte is not a Line Feed as expected.
+     */
+    private static void checkLineFeed(final InputStream input,
+        final ByteArrayOutputStream baos, final Integer position)
+        throws IOException {
+        if (input.read() != '\n') {
+            throw new HttpException(
+                HttpURLConnection.HTTP_BAD_REQUEST,
+                String.format(
+                    "there is no LF after CR in header, line #%d: \"%s\"",
```

Comment:
@mkordas could you please explain why it should be more generic? Here is a private static method called by another private static method so it is internal to the class RqLive and used for a very specific use case it is not meant to be reused elsewhere even internally. If for another task it is needed to make it generic, it can be done for this particular task, here our goal is to do the task in the most efficient manner, don't you agree?